src/components/categories-list/CategoriesList.tsx

```diff
@@ -0,0 +1,77 @@
+import { useEffect, useMemo, useState } from 'react'
+import { AxiosResponse } from 'axios'
+import { useTranslation } from 'react-i18next'
+
+import CardWithLink from '~/components/card-with-link/CardWithLink'
+import CardsList from '~/components/cards-list/CardsList'
+import PageWrapper from '~/components/page-wrapper/PageWrapper'
+import { authRoutes } from '~/router/constants/authRoutes'
+import { categoryService } from '~/services/category-service'
+import { CategoryInterface, ItemsWithCount } from '~/types'
+
+interface CategoriesListProps {
+  query: string
+}
+
+const CategoriesList = ({ query }: CategoriesListProps) => {
+  const [categories, setCategories] = useState<CategoryInterface[]>([])
+  const [visibleCards, setVisibleCards] = useState(4)
+  const params = useMemo(() => ({ name: query }), [query])
+
+  const { t } = useTranslation()
+
+  const fetchCategories = async () => {
+    try {
+      const response: AxiosResponse<ItemsWithCount<CategoryInterface>> =
+        await categoryService.getCategories(params)
+
+      setCategories(response.data.items)
+    } catch (error) {
+      console.error('error', error)
+      setCategories([])
+    }
+  }
+
+  useEffect(() => {
+    void fetchCategories()
+  }, [params])

```

Comment:
move  fetchCategories` into `useEffect`
```suggestion
  useEffect(() => {
    const fetchCategories = async () => {
    try {
      const response: AxiosResponse<ItemsWithCount<CategoryInterface>> =
        await categoryService.getCategories(params)

      setCategories(response.data.items)
    } catch (error) {
      console.error('error', error)
      setCategories([])
    }
  }
    void fetchCategories()
  }, [params])

```